src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/JunctionTreeKBestHaplotypeFinder.java

```diff
@@ -0,0 +1,198 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading.ExperimentalReadThreadingGraph;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading.MultiDeBruijnVertex;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class JunctionTreeKBestHaplotypeFinder<V extends BaseVertex, E extends BaseEdge> extends KBestHaplotypeFinder<V, E> {
+    public static final int DEFAULT_OUTGOING_JT_EVIDENCE_THRESHOLD_TO_BELEIVE = 3;
+    public static final int DEFAULT_MINIMUM_WEIGHT_FOR_JT_BRANCH_TO_NOT_BE_PRUNED = 2;
+    public static final int DEFAULT_MAX_ACCEPTABLE_DECISION_EDGES_WITHOUT_JT_GUIDANCE = 7;
+    public static final int DEFAULT_MAX_ACCEPTABLE_REPETITIONS_OF_A_KMER_IN_A_PATH = 10;
+    private int weightThresholdToUse = DEFAULT_OUTGOING_JT_EVIDENCE_THRESHOLD_TO_BELEIVE;
+
+    // List for mapping vertexes that start chains of kmers that do not diverge, used to cut down on repeated graph traversal
+    Map<V, List<E>> contiguousSequences = new HashMap<>();
```

Comment:
Hmm... I'm going to pin this for now. The chain pruner doesn't quite do the correct thing because we care about also stopping on junction trees or the terminal vertex... Though in principle it shouldn't be that different. I will save this for another iteration for expediency sake.